src/Workspaces/Remote/Core/ProjectSystem/IWorkspaceProject.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using StreamJsonRpc;
+
+namespace Microsoft.CodeAnalysis.Remote.ProjectSystem
+{
+    [RpcMarshalable]
+    public interface IWorkspaceProject : IAsyncDisposable
+    {
+        Task SetDisplayNameAsync(string displayName, CancellationToken cancellationToken);
+
+        Task SetCommandLineArgumentsAsync(IReadOnlyList<string> arguments, CancellationToken cancellationToken);
```

Comment:
curious the reason for these types, and not ImmutableArray?

Reply:
Same as earlier: following best practice docs.